test/test-fs.c

```diff
@@ -2413,6 +2434,9 @@ TEST_IMPL(fs_rename_to_existing_file) {
 
 
 TEST_IMPL(fs_read_file_eof) {
+#if defined(__CYGWIN__) || defined(__MSYS__)
+  RETURN_SKIP("Cygwin pread at EOF may return data.");
```

Comment:
I think it is a bug in Cygwin:

```console
$ cat pread.c
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdio.h>

int main()
{
  char buf[16];
  int fd = open("pread.c", O_RDONLY);
  off_t pos = lseek(fd, 0, SEEK_END);
  ssize_t n = pread(fd, buf, sizeof(buf), pos);
  fprintf(stderr, "n=%d\n", (int)n);
  return 0;
}

$ gcc pread.c && ./a.exe
n=3
```

The same thing on Linux works as expected:

```console
$ gcc pread.c && ./a.out
n=0
```

On Cygwin the bug makes this test case fail.  I'll update the skip reason text to clarify that this is due to incorrect behavior on the platform.